### ISSUE TITLE
rmse calculation

### ISSUE THREAD

**Issue by aimanian** [DhruvJawalkar/Depth-Map-Prediction-from-a-Single-Image-using-a-Multi-Scale-Deep-Network#2]
Hi Dhruv,

I am trying to calculate the root mean square error between the predicted (transformed) depth and and the real (transformed) depth from running your code. I used the following code right after predictions to calculate root mean square error (rmse) between predicted and real depths, but the rmse seems to be very high. I was wondering if you can let me know if I am wrong in rmse calculation. Actually, the rmse reported in the paper is much lower, and I think I am doing a mistake.

Many thanks,

-First I transformed the predictions to 0-255 as you described as,

pred_numpy_arr = pred.detach().squeeze(dim=1).squeeze(dim=0).cpu().numpy()
pred_numpy_arr = pred_numpy_arr - pred_numpy_arr.min() # -0.660375
pred_numpy_arr = (pred_numpy_arr/pred_numpy_arr.max())*255 # max = 2.1548061, brings scale to 0-255
pred_numpy_arr = pred_numpy_arr.astype('uint8')

- Second I used to following code for rmse calculation for batch size of 8 as:
from sklearn.metrics import r2_score
from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error
from math import sqrt
#from math import mean
from sklearn.metrics import r2_score
from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error
from math import sqrt
from scipy.linalg import norm
#from math import mean
import math

r2=[]
mae=[]
b=0
c=0
k=0
bs=8
for i in range(bs):
      a.append((abs(depth.numpy()[i].mean()-pred_numpy_arr[i].mean()))/depth.numpy() . 
      [i].mean())
      r2.append(r2_score(depth.numpy()[i],pred_numpy_arr[i]))
     mae.append((norm(depth.numpy()[i]-pred_numpy_arr[i]))**2/(120*160))
     b=b+a[i]
     c=c+mae[i]
rmse=sqrt(c/8)
rmse

I get rmse=72

**Comment by DhruvJawalkar** [comment]
Your RMSE calculation is incorrect.
Please star the repo if you are using the code !!

You can calculate RMSE directly on the tensors. 
I get values of  ~0.57 for each pixel on the linear scale, which seems to be better than the paper.

See below for the calculations part and you should be fine. 
<img width="455" alt="Screenshot 2019-04-15 at 12 55 54 PM" src="https://user-images.githubusercontent.com/5371976/56161449-60704800-5f7e-11e9-9ed6-68f157d5dc23.png">

**Comment by aimanian** [comment]
Many thanks!
Sure! your code is great! I really enjoyed your approach.
Very nice!

**Comment by aimanian** [comment]
Just one more clarification. I think you should divide the norm calculation by the sqrt of your sample number which means torch.norm(depth-preds)/math.sqrt(8*120*160). This makes a big difference in the RSME value.

**Comment by DhruvJawalkar** [comment]
Hm, yes that seems to be the proper way to calculate. 
I guess then the RMSE is 39.19. Re training for a few more epochs could make improvements
